src/istgt_lu_ctl.c

```diff
@@ -2902,6 +2903,88 @@ _verb_istat ISCSIstat_last[ISCSI_ARYSZ] = { {0,0,0} };
 _verb_istat ISCSIstat_now[ISCSI_ARYSZ] = { {0,0,0} };
 _verb_istat ISCSIstat_rslt[ISCSI_ARYSZ] = { {0,0,0} };
 
+// istgt_uctl_cmd_iostats collects the iostats from the spec structure
+// and marshal them into json format using json-c library.The returned
+// string memory is managed by the json_object and will be freed when
+// the reference count of the json_object drops to zero.
+// TODO: Add the fields for getting the latency, used capacity etc.
+
+static int
+istgt_uctl_cmd_iostats(UCTL_Ptr uctl)
+{
+	ISTGT_LU_Ptr lu;
+	int rc, length;
+	// instantiate json_object from json-c library.
+        struct json_object *jobj;
+	// these are utility variables that will be freed at the end of the
+	// function.
+	char *writes, *reads, *totalreadbytes, *totalwritebytes, *size;
+	int i, j;
+	ISTGT_LU_DISK *spec;
+	ISTGT_Ptr istgt = uctl->istgt;
+	for (i = 0; i < MAX_LOGICAL_UNIT; i++) {
+                lu = (ISTGT_LU_Ptr *) istgt->logical_unit[i];
+                if (lu == NULL)
+                        continue;
+                for (j = 0; j< lu->maxlun; j++) {
+			spec = (ISTGT_LU_DISK *) lu->lun[j].spec;
+			if (spec == NULL)
+				continue;
+			jobj = json_object_new_object();	// create new object
+			json_object *jIQN = json_object_new_string(lu->name);
+			json_object_object_add(jobj, "iqn", jIQN);
+
+			length = snprintf(NULL, 0, "%"PRIu64, spec->writes);	// get the length
+			writes = malloc(length + 1);
+			snprintf(writes, length + 1, "%"PRIu64, spec->writes);	// uint64 to string
+
+			length = snprintf(NULL, 0, "%"PRIu64, spec->reads);
+			reads = malloc(length + 1);
+			snprintf(reads, length + 1, "%"PRIu64, spec->reads);
+
+			length = snprintf(NULL, 0, "%"PRIu64, spec->readbytes);
+			totalreadbytes = malloc(length + 1);
+			snprintf(totalreadbytes, length + 1, "%"PRIu64, spec->readbytes);
+
+			length = snprintf(NULL, 0, "%"PRIu64, spec->writebytes);
+			totalwritebytes = malloc(length + 1);
+			snprintf(totalwritebytes, length + 1, "%"PRIu64, spec->writebytes);
+
+			length = snprintf(NULL, 0, "%"PRIu64, spec->size);
+			size = malloc(length + 1);
+			snprintf(size, length + 1, "%"PRIu64, spec->size);
+
+			json_object *jreads = json_object_new_string(reads);	// instantiate object
+			json_object *jwrites = json_object_new_string(writes);
+			json_object *jtotalreadbytes = json_object_new_string(totalreadbytes);
+			json_object *jtotalwritebytes = json_object_new_string(totalwritebytes);
+			json_object *jsize = json_object_new_string(size);
+
+			json_object_object_add(jobj, "writes", jwrites);	// add values to object field
+			json_object_object_add(jobj, "reads", jreads);
+			json_object_object_add(jobj, "totalwritebytes", jtotalwritebytes);
+			json_object_object_add(jobj, "totalreadbytes", jtotalreadbytes);
+			json_object_object_add(jobj, "size", jsize);
+
+			istgt_uctl_snprintf(uctl, "%s  %s\n", uctl->cmd, json_object_to_json_string(jobj));
+			rc = istgt_uctl_writeline(uctl);
```

Comment:
you need to free JSON object `jobj` here. You can use `json_object_put` here. As `ref-count` for that JSON object reaches to `0`, it will free JSON object.